featuretools/tests/entityset_tests/test_es.py

```diff
@@ -277,101 +277,115 @@ def test_add_relationship_error(es):
-#         df = df.sort_values('id')
+def test_query_by_id_with_time(es):
+    df = es.query_by_values(
+        dataframe_name='log',
+        instance_vals=[0, 1, 2, 3, 4],
+        time_last=datetime(2011, 4, 9, 10, 30, 2 * 6))
+    df = to_pandas(df)
+    if ks and isinstance(es['log'], ks.DataFrame):
+        # Koalas doesn't maintain order
+        df = df.sort_values('id')
 
-#     assert list(df['id'].values) == [0, 1, 2]
+    assert list(df['id'].values) == [0, 1, 2]
 
 
-# def test_query_by_variable_with_time(es):
-#     df = es.query_by_values(
-#         entity_id='log',
-#         instance_vals=[0, 1, 2], variable_id='session_id',
-#         time_last=datetime(2011, 4, 9, 10, 50, 0))
-#     df = to_pandas(df)
+def test_query_by_variable_with_time(es):
+    df = es.query_by_values(
+        dataframe_name='log',
+        instance_vals=[0, 1, 2], column_name='session_id',
+        time_last=datetime(2011, 4, 9, 10, 50, 0))
+    df = to_pandas(df)
 
-#     true_values = [
-#         i * 5 for i in range(5)] + [i * 1 for i in range(4)] + [0]
-#     if ks and isinstance(es['log'].df, ks.DataFrame):
-#         # Koalas doesn't maintain order
-#         df = df.sort_values('id')
+    true_values = [
+        i * 5 for i in range(5)] + [i * 1 for i in range(4)] + [0]
+    if ks and isinstance(es['log'], ks.DataFrame):
+        # Koalas doesn't maintain order
+        df = df.sort_values('id')
 
-#     assert list(df['id'].values) == list(range(10))
-#     assert list(df['value'].values) == true_values
+    assert list(df['id'].values) == list(range(10))
+    assert list(df['value'].values) == true_values
 
 
-# def test_query_by_variable_with_training_window(es):
-#     df = es.query_by_values(
-#         entity_id='log',
-#         instance_vals=[0, 1, 2], variable_id='session_id',
-#         time_last=datetime(2011, 4, 9, 10, 50, 0),
-#         training_window='15m')
-#     df = to_pandas(df)
+def test_query_by_variable_with_no_lti_and_training_window(es):
+    match = "Using training_window but last_time_index is not set for dataframe log"
+    with pytest.warns(UserWarning, match=match):
+        df = es.query_by_values(
+            dataframe_name='log',
+            instance_vals=[0, 1, 2], column_name='session_id',
+            time_last=datetime(2011, 4, 9, 10, 50, 0),
+            training_window='15m')
+    df = to_pandas(df)
+
+    assert list(df['id'].values) == [9]
+    assert list(df['value'].values) == [0]
+
+
+def test_query_by_variable_with_lti_and_training_window(es):
+    es.add_last_time_indexes()
+    df = es.query_by_values(
+        dataframe_name='log',
+        instance_vals=[0, 1, 2], column_name='session_id',
+        time_last=datetime(2011, 4, 9, 10, 50, 0),
+        training_window='15m')
+    df = to_pandas(df)
 
-#     assert list(df['id'].values) == [9]
-#     assert list(df['value'].values) == [0]
+    assert list(df['id'].values) == [9]
+    assert list(df['value'].values) == [0]
```

Comment:
for this test, should we pick an example where the last_time_index impacts the returned values? it would have to be a different dataframe

Reply:
Probably a good idea. Do you think both of the training window tests should use the same dataframe so the difference between having lti set and not is clear?

Reply:
sure 👍

Reply:
@rwedge The two tests that use training window have been updated to produce different results with a last time index set.